docs/source/reference/tpu.rst

```diff
@@ -37,12 +38,72 @@ Like :ref:`GPUs <interactive-nodes>`, SkyPilot provides a simple command to quic
 
 After the command has finished, you will be dropped into the host VM and can start develop code right away!
 
-Below we demonstrate how to run MNIST training on both TPU Nodes and TPU VMs with SkyPilot YAML.
+Below we demonstrate how to run MNIST training on both TPU VMs and TPU Nodes with SkyPilot YAML.
+
+TPU VMs
+--------------------------------
+
+To use TPU VMs, user needs to specify the TPU type, :code:`tpu_vm: True` and the desired TPU runtime version in :code:`accelerator_args` shown below:
+
+.. code-block:: yaml
+
+   resources:
+      accelerators: tpu-v2-8
+      accelerator_args:
+         runtime_version: tpu-vm-base
+         tpu_vm: True
+
+
+Now we show an example of running `mnist training <https://cloud.google.com/tpu/docs/run-calculation-jax#running_jax_code_on_a_tpu_vm>`_ on TPU VM with JAX.
+
+.. code-block:: yaml
+
+   name: mnist-tpu-vm
+
+   resources:
+      accelerators: tpu-v2-8
+      accelerator_args:
+         runtime_version: tpu-vm-base
+         tpu_vm: True
+
+   setup: |
+      git clone https://github.com/google/flax.git
+
+      conda activate flax
+      if [ $? -eq 0 ]; then
+         echo 'conda env exists'
+      else
+         conda create -n flax python=3.8 -y
+         conda activate flax
+         # Make sure to install TPU related packages in a conda env to avoid package conflicts.
+         pip install "jax[tpu]>=0.2.16" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+         pip install --upgrade clu
+         pip install -e flax
+      fi
+
+   run: |
+      conda activate flax
+      cd flax/examples/mnist
+      python3 main.py --workdir=/tmp/mnist \
+      --config=configs/default.py \
+      --config.learning_rate=0.05 \
+      --config.num_epochs=10
+
+A GCS bucket is not required as the TPU VM is physically linked to the TPU device, which can access data directly.
```

Comment:
This sentence is a bit out of the blue. Possibly due to reordering?

Reply:
Fixed. thanks!